// appium-android-driver/build/lib/commands/bidi/models.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.makeObsoleteContextUpdatedEvent = exports.makeContextUpdatedEvent = void 0;
exports.makeLogEntryAddedEvent = makeLogEntryAddedEvent;
const helpers_1 = require("../context/helpers");
const constants_1 = require("./constants");
function toContextUpdatedEvent(method, contextName) {
    return {
        method,
        params: {
            name: contextName,
            type: contextName === helpers_1.NATIVE_WIN ? 'NATIVE' : 'WEB',
        },
    };
}
const makeContextUpdatedEvent = (contextName, domain) => toContextUpdatedEvent((0, constants_1.CONTEXT_UPDATED_EVENT)(domain), contextName);
exports.makeContextUpdatedEvent = makeContextUpdatedEvent;
/**
 * @deprecated Use {@link makeContextUpdatedEvent} instead
 */
const makeObsoleteContextUpdatedEvent = (contextName) => toContextUpdatedEvent(constants_1.CONTEXT_UPDATED_EVENT_OBSOLETE, contextName);
exports.makeObsoleteContextUpdatedEvent = makeObsoleteContextUpdatedEvent;
function makeLogEntryAddedEvent(entry, context, type) {
    return {
        context,
        method: constants_1.LOG_ENTRY_ADDED_EVENT,
        params: {
            type,
            level: entry.level,
            source: {
                realm: '',
            },
            text: entry.message,
            timestamp: entry.timestamp,
        },
    };
}
